File: scripts/examples/pause-when-minimize.cs

// Pauses playback when window is minimized and resumes afterwards.

using System;
using System.Windows.Forms;

using mpvnet;

class Script
{
    MainForm Form;

    bool WasPlaying;
    bool WasPaused;

    public Script()
    {
        Form = MainForm.Instance;
        Form.Resize += Form_Resize;
    }

    private void Form_Resize(object sender, EventArgs e)
    {
        if (Form.WindowState == FormWindowState.Minimized)
        {
            WasPlaying = !mp.get_property_bool("pause");

            if (WasPlaying)
            {
                mp.set_property_bool("pause", true, true);
                WasPaused = true;
            }
        }
        else
        {
            if (WasPaused)
            {
                mp.set_property_bool("pause", false, true);
                WasPaused = false;
            }
        }
    }
}

File: scripts/examples/pause-when-minimize.lua

local did_minimize = false

mp.observe_property("window-minimized", "bool", function(name, value)
    local pause = mp.get_property_bool("pause")

    if value == true then
        if pause == false then
            mp.set_property_bool("pause", true)
            did_minimize = true
        end
    elseif value == false then
        if did_minimize and (pause == true) then
            mp.set_property_bool("pause", false)
        end

        did_minimize = false
    end
end)
